node/network/src/peer.rs

```diff
@@ -91,9 +112,11 @@ pub struct Peer {
     inc_msg_tx: Sender<(PeerId, Vec<u8>)>,
     /// How long do we wait before reconnecting to the peer.
     reconnect_delay: Duration,
+    /// Chain state on peer connection.
+    chain_state_retriever: T,
 }
 
-impl Peer {
+impl<T: ChainStateRetriever + Sized + Send + Clone + 'static> Peer<T> {
```

Comment:
I think it makes sense to apply `+ Sized + Send + 'static` constraints on `ChainStateRetriever` itself.